server/utils/validation/signUpSchema.ts

```diff
@@ -0,0 +1,14 @@
+import Joi from 'joi';
+
+const signUpSchema = async (data: object) => {
+  const schema = Joi.object({
+    name: Joi.string().required(),
+    email: Joi.string().email().required(),
+    password: Joi.string().regex(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{4,}$/).required(),
+  });
+
+  const result: any = await schema.validateAsync(data, { abortEarly: false });
+  return result;
+};
+
+export default signUpSchema;
```

Comment:
When I read signUoSchema I thought it was just the schema and you will do the validate in the controller, so rename this to something more descriptive, and edit the type any.
Also `abortEarly` is by default false so no need to write it.